common/src/main/java/com/wynntils/services/wynnlanguage/WynnLanguageService.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.wynnlanguage;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.components.Service;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.activities.discoveries.DiscoveryInfo;
+import com.wynntils.models.activities.type.ActivitySortOrder;
+import com.wynntils.utils.mc.McUtils;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.item.ItemStack;
+
+public class WynnLanguageService extends Service {
```

Comment:
Since this is about the Wynn world, it should be a Model, not a service.

The respective base classes give these descriptions to help tell them apart:

 * Models are our representation of the Wynncraft world, game state and Wynncraft
 * metadata. In general, any non-trivial interesting aspect of Wynncraft that is
 * provided through vanilla Minecraft elements, should be parsed and provided by
 * a Model.

 * Services provide support to consumers for anything that is not specifically
 * related to the Wynncraft model, either since it is an external service
 * (3rd party library, or system/environment interaction), or since it is something
 * that we augment the Wynncraft world with, but the Wynncraft servers has no knowledge
 * about.

Reply:
Ah didn't know they had a description, should probably look at those base classes

Reply:
can you add that to for example the contribution guide line

Reply:
You can also check out the overall architecture document: https://github.com/Wynntils/Artemis/wiki/Class-and-package-structure

It's not completely up to date (it does not list Services, for instance) but it is a good start.